src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -57,478 +59,270 @@ public ResultTextMarker(int runId, Region region, string fullFilePath)
             Color = DEFAULT_SELECTION_COLOR;
         }
 
-        // This method is called when you click an inline link, with an integer target, which
-        // points to a Location object that has a region associated with it.
-        internal IVsWindowFrame NavigateTo(bool usePreviewPane)
+        /// <summary>
+        /// Attempts to navigate a VS editor to the text marker.
+        /// </summary>
+        /// <param name="usePreviewPane">Indicates whether to use VS's preview pane.</param>
+        /// <returns>Returns true if a VS editor was opened.</returns>
+        public bool TryNavigateTo(bool usePreviewPane)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
-            // Before anything else, see if this is an external link we should open in the browser.
-            Uri uri;
-            if (Uri.TryCreate(this.FullFilePath, UriKind.Absolute, out uri))
+
+            return this.TryNavigateTo(usePreviewPane, retryNaviation: true);
+
+        }
+
+        private bool TryNavigateTo(bool usePreviewPane, bool retryNaviation)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
>ThreadHelper.ThrowIfNotOnUIThread(); [](start = 12, length = 36)

No need for the additional check. The only ways in here are (a) through the public API, which checks, and (b) recursively from this method, so the check has already been done.

Reply:
So if I remove this line the analyzers still complain. Do we need to update the analyzers? (They are in error in this case).

---
In reply to: [482127215](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482127215) [](ancestors = 482127215)

Reply:
Hmm. Can you remove it from the public API? If not, then yes, the analyzer could be improved.

---
In reply to: [482307432](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482307432) [](ancestors = 482307432,482127215)

Reply:
Nope. Trying to remove it from either place makes the analyzers complain. (Ugh)

---
In reply to: [482377108](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482377108) [](ancestors = 482377108,482307432,482127215)